model/metrics/dotnet-metrics.yaml

```diff
@@ -0,0 +1,190 @@
+    stability: experimental
+
+  - id: metric.dotnet.jit.compilation.time
+    type: metric
+    metric_name: dotnet.jit.compilation.time
+    brief: >
+      The amount of time the JIT compiler has spent compiling methods since the
+      process has started.
+    note: "This metric reports the same values as calling [`JitInfo.GetCompilationTime()`](https://learn.microsoft.com/dotnet/api/system.runtime.jitinfo.getcompilationtime)."
+    instrument: counter
+    unit: "s"
+    stability: experimental
+
+  - id: metric.dotnet.monitor.lock_contention.count
+    type: metric
+    metric_name: dotnet.monitor.lock_contention.count
+    brief: >
+      The number of times there was contention when trying to acquire a monitor
+      lock since the process has started.
+    note: "This metric reports the same values as calling [`Monitor.LockContentionCount`](https://learn.microsoft.com/dotnet/api/system.threading.monitor.lockcontentioncount)."
+    instrument: counter
+    unit: "{contention}"
+    stability: experimental
+
+  - id: metric.dotnet.thread_pool.thread.count
+    type: metric
+    metric_name: dotnet.thread_pool.thread.count
+    brief: "The number of thread pool threads that currently exist."
+    note: "This metric reports the same values as calling [`ThreadPool.ThreadCount`](https://learn.microsoft.com/dotnet/api/system.threading.threadpool.threadcount)."
+    instrument: updowncounter
+    unit: "{thread}"
+    stability: experimental
+
+  - id: metric.dotnet.thread_pool.work_item.count
+    type: metric
+    metric_name: dotnet.thread_pool.work_item.count
+    brief: >
+      The number of work items that the thread pool has completed since
+      the process has started.
+    note: "This metric reports the same values as calling [`ThreadPool.CompletedWorkItemCount`](https://learn.microsoft.com/dotnet/api/system.threading.threadpool.completedworkitemcount)."
+    instrument: counter
+    unit: "{work_item}"
+    stability: experimental
+
+  - id: metric.dotnet.thread_pool.queue.length
+    type: metric
+    metric_name: dotnet.thread_pool.queue.length
+    brief: >
+      The number of work items that are currently queued to be processed by the
+      thread pool.
+    note: "This metric reports the same values as calling [`ThreadPool.PendingWorkItemCount`](https://learn.microsoft.com/dotnet/api/system.threading.threadpool.pendingworkitemcount)."
+    instrument: updowncounter
+    unit: "{queue}"
+    stability: experimental
+
+  - id: metric.dotnet.timer.count
+    type: metric
+    metric_name: dotnet.timer.count
+    brief: "The number of timer instances that are currently active."
+    note: "This metric reports the same values as calling [`Timer.ActiveCount`](https://learn.microsoft.com/dotnet/api/system.threading.timer.activecount)."
+    instrument: updowncounter
+    unit: "{timer}"
+    stability: experimental
+
+  - id: metric.dotnet.assemblies.count
+    type: metric
+    metric_name: dotnet.assemblies.count
```

Comment:
this is number of currently loaded assemblies, it's an updown counter and based on the proposal in the https://github.com/open-telemetry/semantic-conventions/pull/1035/files#r1646930420
should be `dotnet.assembly.count`

Reply:
Fixed in latest commits